README.md

```diff
@@ -1,95 +1,104 @@
+### Requirements
 
----
+- Anchor 0.29.0
+- Solana 1.17.22
 
-# Whirlpool SDK
+### Getting Started
 
-Use the SDK to interact with a deployed Whirlpools program via Typescript.
+These instructions are for setting up a development environment on a Mac. If you are using a different operating system, you will need to adjust the instructions accordingly.
 
-## Installation
+* Install NodeJS and gcc-12 using `brew install node gcc@12`.
+* Add gcc-12 headers to your cpath using `export CPATH="/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include"`.
+* Install Rust lang using `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`.
+* Install the Solana CLI using `curl -sSfL https://release.solana.com/v1.17.22/install | sh`.
+* Add the Solana CLI to your path using `export PATH="/Users/$(whoami)/.local/share/solana/install/active_release/bin:$PATH"`.
+* Install Anchor version manager using `cargo install --git https://github.com/coral-xyz/anchor avm --locked --force`.
+* Install the latest Anchor version using `avm install 0.29.0 && avm use 0.29.0`.
+* Clone this repository using `git clone https://github.com/jewl-app/.github`.
+* Install the dependencies using `yarn`.
+* Set up a Solana wallet if you don't have one already (see below).
+* Run one of the commands below to get started such as `yarn build`.
 
-In your package, run:
+#### Setting up a Solana wallet
 
-```
-yarn add "@orca-so/whirlpools-sdk"
-yarn add "@orca-so/common-sdk"
-yarn add "@coral-xyz/anchor@0.29.0"
-yarn add "@solana/web3.js"
-yarn add "@solana/spl-token"
-yarn add "decimal.js"
-```
+* Create a new keypair using `solana-keygen new`.
+* Check if you have a valid wallet address using `solana address`.
+* Set your local config to the Solana devnet env using `solana config set --url https://api.devnet.solana.com`.
+* Give yourself some devnet SOL (for transaction fees) using `solana airdrop 1`.
+* Check if you have a positive balance using `solana balance`.
 
-## Usage
+### Components
 
-Read instructions on how to use the SDK on the [Orca Developer Portal](https://orca-so.gitbook.io/orca-developer-portal/orca/welcome).
+This repository uses NX to manage the Rust and Typescript codebases. This allows us to have a monorepo with multiple packages and share code between them. Dependencies between packages are automatically resolved by NX, so you don't have to worry about managing that yourself.
 
-## Run Typescript tests via local validator
+This repository is split up into sevaral parts. The following is a (non-exhaustive) list of the components and their purpose.
 
-In the whirlpools/sdk folder, run:
+* **`/programs/*`** - Rust programs that are deployed on Solana.
+* **`/typescript/*`** (WIP) - Typescript SDKs for interacting with the programs.
+* **`/rust/*`** (WIP) - Rust SDKs for interacting with the programs.
+* **`/docs/*`** (WIP) - Documentation for the programs and SDKs.
+* **`/legacy/*`** - Legacy Typescript SDKs and integration tests.
```

Comment:
I believe that `/typescript`, `/rust` and `/legacy`  are all SDK and only `/program` is contract.
So WDYT the following:

- /docs
- /program
- /sdk
    - /legacy
    - /typescript
    - /rust

Reply:
Think this is similar to how anza does their structure. I am not the biggest fan of the extra nesting since `legacy`, `typescript` and `rust` each contain multiple package. If you feel strongly about this one I can change

Reply:
I think `/sdk` is immediately recognisable as an SDK to devs who come looking for an SDK.
The location of the SDK is clear if they come via the Developer Portal, and the README.md is also useful. However, it would be nice if the structure itself is descriptive, as there are many cases where people do not read the documentation (including me).😂

If this change would significantly affect subsequent PRs, I agree to merge this PR as-is for the time being, as it can be changed after all the current PRs have been merged.

Reply:
We can also go for 'legacy-sdk', 'rust-sdk' and 'ts-sdk' so we don't have the extra nesting. What do you think?

Reply:
But that might also be a little weird cause in #172 it would then add 'sdk/rust/sdk' and 'sdk/rust/client' or 'rust-sdk/sdk' etc.

Reply:
@wjthieme 
Yep, I agree that `x-sdk` naming is nice alternative.
Let's go with it 🚀